File: rb/features.rb
# Fastcon SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FastconFeatures
  def self.make_feature(name)
    case name
    when "base"
      FastconBaseFeature.new
    when "ratelimit"
      FastconRatelimitFeature.new
    when "retry"
      FastconRetryFeature.new
    when "test"
      FastconTestFeature.new
    when "timeout"
      FastconTimeoutFeature.new
    else
      FastconBaseFeature.new
    end
  end
end

File: py/features.py
# Fastcon SDK feature factory

from feature.base_feature import FastconBaseFeature
from feature.test_feature import FastconTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FastconBaseFeature(),
        "test": lambda: FastconTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
